openeogeotrellis/load_stac.py

```diff
@@ -84,9 +84,28 @@ def supports_item_search(coll: pystac.Collection) -> bool:
         conforms_to = coll.get_root().extra_fields.get("conformsTo", [])
         return any(conformance_class.endswith("/item-search") for conformance_class in conforms_to)
 
+    def is_raster_mime_type(mime_type: str) -> bool:
+        mime_type = mime_type.lower()
+        # https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#common-media-types-in-stac
+        if (
+            mime_type.startswith("image/tif")  # match both tif and tiff, just in case
+            or mime_type.startswith("image/vnd.stac.geotiff")
+            or mime_type.startswith("image/jp2")
+            or mime_type.startswith("image/png")
+            or mime_type.startswith("image/jpeg")
+            or mime_type.startswith("application/x-hdf")  # matches hdf5 and hdf
+            or mime_type.startswith("application/x-netcdf")
+            or mime_type.startswith("application/netcdf")
+        ):
+            return True
+
     def is_band_asset(asset: pystac.Asset) -> bool:
         roles_with_bands = ["data", "data-mask", "snow-ice", "land-water", "water-mask"]
-        return any(asset.has_role(role) for role in roles_with_bands) or "eo:bands" in asset.extra_fields
+        return (
+            any(asset.has_role(role) for role in roles_with_bands)
+            and (asset.media_type is None or is_raster_mime_type(asset.media_type))
+            or "eo:bands" in asset.extra_fields
+        )
```

Comment:
```suggestion
        return (
            (any(asset.has_role(role) for role in roles_with_bands) or "eo:bands" in asset.extra_fields)
            and (asset.media_type is None or is_raster_mime_type(asset.media_type))
        )
```

- builds upon previous implementation
- otherwise might still want to load unsupported rasters
- parentheses for clarity

Reply:
After changing this, I also had to add `application/octet-stream` to the list for a test running on https://catalogue.dataspace.copernicus.eu/stac/collections/GLOBAL-MOSAICS 
A bit annoying, because that stac catalog is quite broken,

Reply:
No, that's what I meant (elsewhere on this page): adapt the test rather than the code. 

We're never going to be able to load that collection in its current state so don't bother.

Reply:
Ah, only now I see the comment. Removed `application/octet-stream` indeed and adapted `test_load_stac_issue830_alternate_url_s3`